### ISSUE TITLE
Go through the proposal text

### ISSUE THREAD

**Issue by monoidal** [tweag/ghc#508]
The goal of this ticket is go to through the text of the [proposal](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0111-linear-types.rst) once more, and make sure all deficiencies have a matching ticket.

Once this is done, let's close https://gitlab.haskell.org/ghc/ghc/-/issues/15981.

**Comment by aspiwack** [comment]
I don't think that we ought to close until we are indeed done with the proposal. That is, the ticket your are, rightly, suggesting we write, ought to be seen as subtasks of 15981.
